src/docfx/build/manifest/ManifestBuilder.cs

```diff
@@ -48,33 +37,34 @@ public bool TryAdd(Document file, FileManifest manifest)
                 return false;
             }
 
+            if (monikers.Count == 0)
+            {
+                monikers = new List<string> { "NONE_VERSION" };
+            }
+            _filesBySiteUrl.GetOrAdd(manifest.SiteUrl, _ => new ConcurrentDictionary<Document, List<string>>()).TryAdd(file, monikers);
+
             return true;
         }
 
         public (FileManifest[], List<Document>) Build(Context context)
         {
-            // Handle publish conflicts
-            foreach (var (siteUrl, conflict) in _siteUrlConflicts)
+            // Handle publish url conflicts
+            foreach (var (siteUrl, files) in _filesBySiteUrl)
             {
-                var conflictingFiles = new HashSet<Document>();
-
-                foreach (var conflictingFile in conflict)
+                var conflictMoniker = files
```

Comment:
Still think we should output to the user which files are causing the conflicts, add a TODO for it?